app/services/workflow_executions/preparation_service.rb

```diff
@@ -11,12 +11,22 @@ class PreparationService < BaseService
     def initialize(workflow_execution, user = nil, params = {})
       super(user, params)
       @workflow_execution = workflow_execution
-      @samplesheet_headers = parse_samplesheet_headers
+      @pipeline = find_pipeline
       @samplesheet_rows = []
       @storage_service = ActiveStorage::Blob.service
     end
 
-    def execute # rubocop:disable Metrics/MethodLength
+    def execute # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+      # confirm pipeline found
+      if @pipeline.nil?
+        @workflow_execution.state = :error
+        @workflow_execution.cleaned = true
+        @workflow_execution.save
+        return false
+      end
```

Comment:
Due to now needing an additional rubocop disable, I think we can just move checking `@pipeline.nil?`  into its own method.

Can you replace this with `return false unless validate_pipeline`

And then under `private`:
```
def validate_pipeline
      return true unless @pipeline.nil?

      @workflow_execution.state = :error
      @workflow_execution.cleaned = true
      @workflow_execution.save
      false
    end
```